crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -280,40 +280,61 @@ impl Starknet {
     /// # Arguments
     ///
     /// * `transaction` - Transaction to be added in the collection of transactions.
+    /// * `contract_class` - Contract class to be added in the state cache. Only in declare
+    ///   transactions.
     /// * `transaction_result` - Result with transaction_execution_info
     pub(crate) fn handle_transaction_result(
         &mut self,
         transaction: Transaction,
+        contract_class: Option<ContractClass>,
```

Comment:
Instead of forcing all transaction types to send provide an argument here, is there an alternative?

Reply:
Only came to mind is to extend the DeclareTransaction to include this property, but it would have end up with far more changes.